addressed a better differentiation between "combining roles" and "topological models" (#77)
ietf-rats-wg/architecture: draft-ietf-rats-architecture.md

In this situation, the trust model described in {{trustmodel}} is also suitable for this inside Verifier.

> ## Roles and Entities

Reviewer: My opinion: I don't like trying to make "Entities" some formal term.   I suspect "implementation" works in most cases in the text below

Reviewer 2: Agree that "Entity" isn't needed in the section heading. But it is helpful to separate the role from the thing that implements the role. Entity is the word most commonly used in the arch to refer to this thing. So it is OK to use it here.

Author: There are various uses of entities throughout the document, including composite device. We decided against "device" in favor of "entity" and in the context of this document the use of this term is now well established. We do not want to provide implementation guidance here, but an architectural concept "on how roles can be collapsed" which is important to know to understand for the solution I-Ds because it allows for the omission of internet protocols.

I see that Dave and Ned agree, but this is one of the rare cases in which I am not convinced. I actually think a focus on implementation guidance distracts from the important parts (such as reuse of key material across roles, etc.).

Reviewer: When did we "decide against device in favor of entity"?   I don't think I agree with that.

> An entity in the RATS architecture includes at least on of the roles defined in this document. As a result, the entity can participate as a constituent of the RATS architecture. Additionally, an entity can aggregate more than one role into itself. These collapsed roles combine the duties of multiple roles. In these cases, interaction between these roles do not necessarily use the Internet Protocol. They can, e.g. using a loopback device or other IP-based communication between separate environments, but they do not have to. Alternative channels to convey conceptual messages include sockets, GPIO interfaces, local busses, or hypervisors. This conveyance is sometime referred to as "local conveyance". This type of conveyance is typically found in Composite Devices. Most importantly, these conveyance methods are out-of-scope of RATS, but they are presumed to exist in order to convey conceptual messages appropriately between roles.

Reviewer: Line is too long in the markdown, please split into multiple source lines to make diffs/commenting easier.

> An entity in the RATS architecture includes at least on of the roles defined in this document. As a result, the entity can participate as a constituent of the RATS architecture. Additionally, an entity can aggregate more than one role into itself. These collapsed roles combine the duties of multiple roles. In these cases, interaction between these roles do not necessarily use the Internet Protocol. They can, e.g. using a loopback device or other IP-based communication between separate environments, but they do not have to. Alternative channels to convey conceptual messages include sockets, GPIO interfaces, local busses, or hypervisors. This conveyance is sometime referred to as "local conveyance". This type of conveyance is typically found in Composite Devices. Most importantly, these conveyance methods are out-of-scope of RATS, but they are presumed to exist in order to convey conceptual messages appropriately between roles.

Reviewer: typo: "on" -> "one"

Author: fixed

> An entity in the RATS architecture includes at least on of the roles defined in this document. As a result, the entity can participate as a constituent of the RATS architecture. Additionally, an entity can aggregate more than one role into itself. These collapsed roles combine the duties of multiple roles. In these cases, interaction between these roles do not necessarily use the Internet Protocol. They can, e.g. using a loopback device or other IP-based communication between separate environments, but they do not have to. Alternative channels to convey conceptual messages include sockets, GPIO interfaces, local busses, or hypervisors. This conveyance is sometime referred to as "local conveyance". This type of conveyance is typically found in Composite Devices. Most importantly, these conveyance methods are out-of-scope of RATS, but they are presumed to exist in order to convey conceptual messages appropriately between roles.

Reviewer: Always put a comma after e.g.

Author: fixed

> An exemplary usage scenario: An entity that both connects to a wide-area network and to a system bus taking on both the Attester and Verifier roles. As a system bus entity, a Verifier consumes Evidence from other devices connected to the system bus that implement Attester roles. As a wide-area network connected entity, it may implement an Attester role. The entity, as a system bus Verifier, may choose to fully isolate its role as a wide-area network Attester. Alternatively, attestation results might reflect an aggregation of system bus operational state claims in the form of Evidence.

Reviewer: "attestation results" -> "Attestation Results"

Author: After "staring" at the diff a bit, attestation results -> Attestation Results is now fixed. Did I miss something?

> An exemplary usage scenario: An entity that both connects to a wide-area network and to a system bus taking on both the Attester and Verifier roles. As a system bus entity, a Verifier consumes Evidence from other devices connected to the system bus that implement Attester roles. As a wide-area network connected entity, it may implement an Attester role. The entity, as a system bus Verifier, may choose to fully isolate its role as a wide-area network Attester. Alternatively, attestation results might reflect an aggregation of system bus operational state claims in the form of Evidence.

Reviewer: "An exemplary usage scenario:" is hard to read for me since it's not a sentence or a section heading or a dictionary entry

Author: "The following paragraph elaborates on an exemplary usage scenario:" is now followed by a paragraph.

> An exemplary usage scenario: An entity that both connects to a wide-area network and to a system bus taking on both the Attester and Verifier roles. As a system bus entity, a Verifier consumes Evidence from other devices connected to the system bus that implement Attester roles. As a wide-area network connected entity, it may implement an Attester role. The entity, as a system bus Verifier, may choose to fully isolate its role as a wide-area network Attester. Alternatively, attestation results might reflect an aggregation of system bus operational state claims in the form of Evidence.

Reviewer 2: I'm experiencing cognitive dissonance trying to relate the roles to both a bus connected example and a network connected example. It seems to be focusing on the composite device but highlighting the case where composite attesters could be network connected. However, I'm not certain based on this text. But otherwise, why wouldn't that discussion belong in the section on composite device?

I think the goal is to illustrate use cases where roles can be combined. Maybe sticking to a few intuitive examples is best. e.g., verifier + relying party, endorser + verifier and maybe verifier+owner. Combining attester and verifier is problematic because the trust model objective is to separate attester from verifier.

Author: TBH, I just copied your description from #73 into here. I probably set the wrong context?

Author: I assume line 401 is okay?

Reviewer 2: Do you mean line 481?
I think everything after "interaction between these roles do not necessarily use the Internet Protocol" is not about role interactions but entity interactions. I don't know if entity interactions belongs or if it does should be a separate section or at least separate paragraph. The main point of line/paragraph 481 is that an entity can host (combine) multiple roles. When that happens, the role messages between roles on peer entities remains true to the roles architecture. For example, if Verifier and RP are combined. Evidence still flows to Verifier and RP still consumes Appraisal policy for Attestation Results (even though there isn't an Attestation Results flow exposed externally to the entity).

Author: It remains an interaction between roles, the means are changing though. Even if all roles would be collapsed, they'd still interact, "locally". Nothing changes architecturally, I think. I consequence, this is not intended to be about entity interaction. Maybe we should simply spell that out?

the RATS architecture. Additionally, an entity can aggregate more than one
role into itself. These collapsed roles combine the duties of multiple roles.
In these cases, interaction between these roles do not necessarily use the
Internet Protocol. They can be using a loopback device or other IP-based
communication between separate environments, but they do not have to.
Alternative channels to convey conceptual messages include function calls, sockets, GPIO
interfaces, local busses, or hypervisor calls. This type of conveyance is typically found
in Composite Devices. Most importantly, these conveyance methods are
out-of-scope of RATS, but they are presumed to exist in order to convey
conceptual messages appropriately between roles.

For example, an entity that both connects to a wide-area network and to a system bus is taking on both the Attester and Verifier roles.
As a system bus entity, a Verifier consumes Evidence from other devices
connected to the system bus that implement Attester roles. As a wide-area
network connected entity, it may implement an Attester role. The entity, as a
system bus Verifier, may choose to fully isolate its role as a wide-area

> network Attester. Alternatively, Attestation Results might reflect an
> aggregation of system bus operational state claims in the form of Evidence.

Reviewer 3: ```suggestion
network Attester. 
```

~~~~
{: #combination title="Example Combination"}

# Roles and Entities

An entity in the RATS architecture includes at least one of the roles defined
in this document. As a result, the entity can participate as a constituent of
the RATS architecture. Additionally, an entity can aggregate more than one
role into itself. These collapsed roles combine the duties of multiple roles.
In these cases, interaction between these roles do not necessarily use the
Internet Protocol. They can be using a loopback device or other IP-based
communication between separate environments, but they do not have to.
Alternative channels to convey conceptual messages include function calls, sockets, GPIO

> interfaces, local busses, or hypervisor calls. This conveyance is sometime
> referred to as "local conveyance". This type of conveyance is typically found

Reviewer 3: ```suggestion
interfaces, local busses, or hypervisor calls. This type of conveyance is typically found
```

~~~~
{: #combination title="Example Combination"}

# Roles and Entities

An entity in the RATS architecture includes at least one of the roles defined
in this document. As a result, the entity can participate as a constituent of
the RATS architecture. Additionally, an entity can aggregate more than one
role into itself. These collapsed roles combine the duties of multiple roles.
In these cases, interaction between these roles do not necessarily use the

> Internet Protocol. They can, e.g., using a loopback device or other IP-based

Reviewer 3: ```suggestion
Internet Protocol. They can be using a loopback device or other IP-based
```

communication between separate environments, but they do not have to.

> Alternative channels to convey conceptual messages include sockets, GPIO

Reviewer 3: ```suggestion
Alternative channels to convey conceptual messages include function calls, sockets, GPIO
```

> Alternative channels to convey conceptual messages include sockets, GPIO

Reviewer: What about a simple function call within the same process?

> interfaces, local busses, or hypervisors. This conveyance is sometime

Reviewer 3: ```suggestion
interfaces, local busses, or hypervisor calls. This conveyance is sometime
```

> referred to as "local conveyance". This type of conveyance is typically found

Reviewer: I'm not sure k we need to define a term for this.

in Composite Devices. Most importantly, these conveyance methods are
out-of-scope of RATS, but they are presumed to exist in order to convey
conceptual messages appropriately between roles.

> The following paragraph elaborates on an exemplary usage scenario:

Reviewer: I still find the sentence above odd, rather than combining an intro sentence into the paragraph below.

> The following paragraph elaborates on an exemplary usage scenario:
> 
> An entity that both connects to a wide-area

Reviewer 3: ```suggestion
For example, an entity that both connects to a wide-area
```

> network and to a system bus taking on both the Attester and Verifier roles.

Reviewer: Above is not a complete sentence

> As a system bus entity, a Verifier consumes Evidence from other devices

Reviewer: Is this referring to a Composite Device example?  If so, move it to that section, or maybe it's already covered there.

Reviewer 3: Could be, but that's not the point.

> The following paragraph elaborates on an exemplary usage scenario:
> 
> An entity that both connects to a wide-area
> network and to a system bus taking on both the Attester and Verifier roles.
> As a system bus entity, a Verifier consumes Evidence from other devices

Reviewer 3: ```suggestion
For example, an entity that both connects to a wide-area network and to a system bus is taking on both the Attester and Verifier roles.
As a system bus entity, a Verifier consumes Evidence from other devices
```
